133 feat people endpoint

SirAntonySir opened the pull request:
Created a GraphQl Endpoint for the People, including People Model, Enums, router, grapgql queries etc.

SirAntonySir committed:
initialize the people by faculty endpoint

SirAntonySir committed:
- Introduced new data models for Person, PersonRole, PersonCourse, and PersonBasicInfo to encapsulate detailed information about individuals.
- Added endpoints for retrieving people lists, filtering by faculty, and fetching individual person details.
- Implemented a data collection service to gather and store people data from the LSF system, including roles and courses.
- Enhanced the database schema with new tables for storing people, their roles, and courses, ensuring a structured approach to data management.

SirAntonySir committed:
- Enhanced the PeopleService to collect and store people data role by role, improving data processing efficiency.
- Introduced crawl_single_role method in LSFPersonCrawler to fetch individuals associated with a specific role ID, including detailed logging for better tracking.
- Updated person ID generation to utilize personal.pid from profile URLs for improved uniqueness.
- Refactored upsert methods to prevent duplication of roles and courses in the database.

SirAntonySir committed:
Refactor LSFPersonCrawler and PeopleService to enhance deduplication and error handling

- Set REQUEST_DELAY to 0 in LSFPersonCrawler to optimize request timing.
- Implemented deduplication for courses and processed people in LSFPersonCrawler to prevent duplicate entries.
- Improved error handling in PeopleService for database operations, specifically for duplicate key violations, ensuring smoother processing of person data.
- Enhanced role and course ID generation to include person-specific identifiers, reducing collision risks.

SirAntonySir committed:
Enhance People API to support faculty ID filtering

- Added `faculty_id` field to `Person` and `PersonSummary` models for better identification.
- Updated `get_people` endpoint to filter by `faculty_id` instead of faculty enum.
- Introduced a new endpoint `get_people_by_faculty_id` for retrieving people by specific faculty ID.
- Enhanced `PeopleService` to map faculty ID to German faculty names and retrieve associated IDs from the database.
- Refactored existing faculty filtering logic to accommodate both faculty enum and ID.

SirAntonySir committed:
Refactor People API endpoints for improved pagination and faculty filtering

- Updated `get_people` endpoint to apply pagination only when no faculty filter is provided.
- Enhanced `get_people_by_faculty_enum` to disable pagination, returning all results for a specific faculty.
- Introduced `get_faculty_enum_by_id` method in `PeopleService` to retrieve faculty enums based on faculty ID.
- Improved error handling and query logic in `PeopleService` for better performance and clarity.

SirAntonySir committed:
- Removed `faculty` and `faculty_id` fields from `Person` and `PersonSummary` models, replacing them with a more streamlined approach using `faculty_enum`.
- Updated `get_people_by_faculty_enum` endpoint to accept `faculty_code` instead of `faculty` enum, enhancing clarity and usability.
- Introduced a new endpoint `get_people_by_faculty_id` for retrieving people based on faculty ID.
- Refactored `PeopleService` methods to utilize faculty codes, improving database queries and error handling.
- Adjusted the `FacultyTable` and `FacultyTranslationTable` to use integer IDs instead of strings for better consistency.

SirAntonySir committed:
delete database services

SirAntonySir committed:
Refactor People API and Data Fetcher for improved structure and functionality

- Consolidated person-related models by importing from a shared module, enhancing code maintainability.
- Updated `people_router` to utilize a new `PeopleAPIService`, streamlining service interactions.
- Enhanced error handling in faculty filtering logic within the `get_people` endpoint.
- Replaced the deprecated `PeopleCollector` with `PeopleCollectorV2` in the data fetcher, improving data collection efficiency.
- Introduced new logging capabilities for better tracking of data fetching processes.
- Refactored enums for academic titles and roles to improve clarity and consistency across the application.

SirAntonySir committed:
Refactor People API to streamline service interactions and enhance response models

- Updated `people_router` to utilize `PeopleService`, simplifying data fetching logic.
- Modified `get_people` endpoint to support filtering by faculty code and improved pagination handling.
- Removed deprecated GraphQL mutation and query files, consolidating functionality within the API.
- Enhanced `Person` model to allow optional basic info and introduced new response models for better data structure.

SirAntonySir committed:
Refactor People GraphQL API and models for improved structure and functionality

- Updated GraphQL mutations to streamline person creation and updates, enhancing clarity and consistency.
- Removed deprecated queries and consolidated functionality within the API, improving maintainability.
- Introduced new response models for person details, roles, and courses, ensuring a more structured data representation.
- Enhanced the `people_router` to utilize updated models, improving data fetching and response handling.
- Adjusted logging levels and error handling in data collection processes for better tracking and debugging.

rubenmaria committed:
feat: defined tables and trying to get example insertion to work in service.

SirAntonySir committed:
Remove legacy test script for PeopleService CMS integration and update GraphQL queries and models for consistency

- Deleted the `test_people_cms_service.py` file as it is no longer needed.
- Updated GraphQL mutations and queries to replace `institution` with `institution_name` for clarity.
- Refactored `PeopleService` and related models to ensure consistent naming conventions across the application.
- Enhanced error handling in the DirectusService for improved logging of GraphQL errors.

SirAntonySir committed:
Enhance People API and Data Fetcher for improved course handling and data normalization

- Updated GraphQL mutations and queries to include courses in person details, streamlining data representation.
- Refactored `PeopleCollectorV2` to remove test mode and improve logging for data collection processes.
- Simplified course mapping in `PeopleModelMapper` to store course numbers directly, enhancing data structure.
- Improved error handling in `PeopleService` for person creation and updates, ensuring robust data processing.
- Consolidated course handling in person details, allowing for better integration with the overall data model.

rubenmaria committed:
feat: can now store lectures in db with sql

rubenmaria committed:
added new endpoint

rubenmaria committed:
bug: lsf updated need to fix crawler to work over mutliple semester

SirAntonySir committed:
Implement role management mutations in GraphQL and enhance PeopleCollectorV2 for async processing

- Added new GraphQL mutations for updating and deleting person roles, improving role management capabilities.
- Refactored `PeopleCollectorV2` to streamline character processing and introduced asynchronous crawling for better performance.
- Enhanced logging for role processing and character crawling, providing clearer insights into data collection activities.
- Improved error handling in role processing to ensure robust data management.

SirAntonySir committed:
Refactor PeopleModelMapper and PersonComplete model for enhanced data handling

- Simplified the creation of `PersonDetails` in `PeopleModelMapper`, consolidating fields and improving clarity.
- Updated the `PersonComplete` model to include properties for email, phone, address, profile URL, office hours, status, and note, allowing for easier access to these details from the `PersonDetails`.
- Enhanced role mapping to store all institutions associated with a role, improving data representation and integrity.

SirAntonySir committed:
Update processing plan in PeopleCollectorV2 to include full character set

rubenmaria committed:
fix: collector now works as expected with lsf update

rubenmaria committed:
feat: added winter semester again

SirAntonySir committed:
Enhance People API with new GraphQL endpoints for faculty and person details

- Introduced `get_people_by_faculty` and `get_person_with_details` endpoints to fetch data using GraphQL queries, improving data retrieval capabilities.
- Updated `DirectusService` to enhance error handling and logging for GraphQL requests, ensuring better tracking of API interactions.
- Retained legacy endpoints for backward compatibility while refactoring the data fetching logic for improved clarity and maintainability.

SirAntonySir committed:
Remove legacy test script and duplicate tracking script; update GraphQL queries for improved consistency

- Deleted `test_api_endpoints.py` and `reset_people_duplicates.py` as they are no longer needed.
- Modified GraphQL queries in `new_people_queries.graphql` to remove unnecessary `limit` parameters, enhancing clarity and consistency.
- Updated various service files to reflect the removal of `limit` parameters, streamlining data fetching logic.

SirAntonySir committed:
Refactor People API by removing legacy GraphQL queries and consolidating service logic

- Deleted `new_people_queries.graphql` as it contained outdated GraphQL queries.
- Updated `people_router.py` to utilize the new `PeopleService`, enhancing data fetching logic.
- Removed `people_directus_service.py` to streamline service architecture and improve maintainability.
- Enhanced `people_service.py` with comprehensive methods for fetching and managing people data, ensuring better integration with the CMS.

SirAntonySir committed:
Refactor PeopleService methods for improved clarity and consistency

- Renamed service methods to private (e.g., `get_people_roles` to `_get_people_roles`) to indicate internal usage.
- Simplified course data handling by directly using the course list instead of converting course data.
- Updated person details fetching method to align with the new naming convention, enhancing code maintainability.

SirAntonySir committed:
Refactor GraphQL queries and service methods for improved clarity and organization. and keep directus service untouched.

- Removed outdated comments and simplified GraphQL query handling in `people_queries.graphql`.
- Renamed query files in `people_service.py` for better clarity and consistency.
- Updated method calls to eliminate unnecessary operation names, enhancing code readability and maintainability.

SirAntonySir committed:
Merge branch '122-feat-create-endpoint-in-backend-for-people' into 133-feat-people-endpoint

SirAntonySir committed:
remove logger and reimport the lecture middleware

SirAntonySir committed:
remove datafetcher components from endpoint pr

rpgraffi commented on api/src/v1/people/routers/people_router.py:
```diff
@@ -0,0 +1,77 @@
+
+
+# Legacy endpoints (keeping for backward compatibility)
+@router.get("/", response_model=PeopleResponse)
```
delete endpoint

rpgraffi commented on api/src/v1/people/routers/people_router.py:
```diff
@@ -0,0 +1,77 @@
+@router.get("/faculty/{faculty_id}")
+async def get_people_by_faculty(
+    faculty_id: int = Path(..., description="Faculty ID to filter by"),
+    offset: int = Query(0, ge=0, description="Number of people to skip")
```
remove offset parameter

rpgraffi commented on api/src/v1/people/routers/people_router.py:
```diff
@@ -0,0 +1,77 @@
+            offset=offset
+        )
+        
+        return {
```
return only people as list

rpgraffi commented on api/src/v1/people/routers/people_router.py:
```diff
@@ -0,0 +1,77 @@
+            "offset": offset
+        }
+        
+    except Exception as e:
```
error handling in Service

rpgraffi commented on shared/src/models/people_model.py:
nothing should be in shared

rpgraffi committed:
(chore): cleanup old tables that moved to directus

rpgraffi committed:
Revert "(chore): cleanup old tables that moved to directus"

This reverts commit 282f4c4ca377738fe6e8978d2c7e28c09b94ad6d.

rpgraffi committed:
chore: remove university-related tables and services from the data fetcher

rpgraffi committed:
Revert "chore: remove university-related tables and services from the data fetcher"

This reverts commit 3608e31f591168f191eb77eb90f67ee2bd029527.

rpgraffi committed:
chore: refactor university-related models and services to streamline data fetching

rpgraffi committed:
chore: remove unused university-related services and clean up screening service

rpgraffi committed:
fix: correct eager loading for university relationships in screening service

rpgraffi committed:
LAST FIX

rpgraffi committed:
LAST LAST FIX

rpgraffi committed:
(feat): prepare prod for backend release and db migration

rpgraffi committed:
(fix): data_fetcher university model

SirAntonySir committed:
Refactor People API to enhance GraphQL queries and service methods

- Updated `get_person_by_id.graphql` to include faculty details as an object instead of a string.
- Refactored `get_people_by_faculty` endpoint in `people_router.py` to return a list of people without offset handling.
- Simplified `get_all_people` method in `people_service.py` to support faculty filtering and improved query handling.
- Adjusted `Person` model to remove title and faculty_enum fields for better alignment with the updated data structure.
- Enhanced academic title handling in `people_enums.py` for more flexible matching.
- Updated Dockerfile to include the API folder for GraphQL file access.

SirAntonySir committed:
remove limiting in api calls remove people model from shared folder

add crawler

SirAntonySir committed:
Refactor People models and services for improved clarity and functionality

- Removed the legacy `PersonBasic` and `PersonComplete` models, consolidating into a single `Person` model.
- Updated `get_all_people` method in `people_service.py` to return a list of `PersonSummary` instead of `PeopleResponse`.
- Enhanced data handling in `people_router.py` to streamline faculty filtering and response formatting.
- Adjusted properties in the `Person` model to directly access details, improving data encapsulation.
- Cleaned up unused imports and comments across various files for better code maintainability.

rpgraffi committed:
(feat): add support for library images added though directus, loaded over sql in code

rubenmaria committed:
feat: added new route for course detail page

dependabot[bot] committed:
chore(deps): bump urllib3 from 2.2.2 to 2.5.0 (#138)

Bumps [urllib3](https://github.com/urllib3/urllib3) from 2.2.2 to 2.5.0.
- [Release notes](https://github.com/urllib3/urllib3/releases)
- [Changelog](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst)
- [Commits](https://github.com/urllib3/urllib3/compare/2.2.2...2.5.0)

---
updated-dependencies:
- dependency-name: urllib3
  dependency-version: 2.5.0
  dependency-type: direct:production
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

rubenmaria committed:
fix: now using existing html to markdown function

rpgraffi committed:
refactor: move html_to_markdown utility to shared.utils and update imports across modules

rubenmaria committed:
fix: now inserting batch-wise into db and using iterator for crawler class

SkyHorizon3 committed:
feat: make recurring events relative to today (#144)

* feat: generate recurring events relative to today

* fix: remove redundant fix

* fix: relativedelta

* refactor: simplify, use offset as recurrence_id

* refactor: remove useless helper func

rpgraffi committed:
(chore): removed links v2 and changed get to post for toggle like

rpgraffi committed:
(bug): removed api endpoint from main

rpgraffi committed:
(fix): add v2 benefits endpoint (moved out of links)

rpgraffi committed:
(chore:) removed arm build from pipeline

rpgraffi committed:
(chore): update mock timeline events

rpgraffi committed:
(refactor): remove wishlist functionality and update link resource structure (#145)

- Deleted wishlist models, services, and routers to streamline the API.
- Updated link resource handling by changing types from LinkType to string.
- Refactored database relationships and imports for links and benefits.
- Removed deprecated link collector and constants related to benefits and resources.

rpgraffi committed:
(feat): likes for links

rpgraffi committed:
(bug): fixed sport crawler by supporting new uninow page

rpgraffi committed:
(bug): fixed tum movie crawler

rubenmaria committed:
📦 [REFACTOR]: Rename all lecture-related code to courses  and removed unnecessary code (#147)

* renamed all symbols, tables and files from class and lecture to course

* removed deadcode and removed unessary commit in database insertion

SkyHorizon3 committed:
feat: improved exception system, improved iCal support (#146)

* feat: more robust and improved exception system

* fix: add missing log message

* refactor: cleanup router

* refactor: change log level

* refactor: update calendar API endpoints and clean up static file mounts

* feat: more safety checks

* fix: endpoint names

* fix: typo

* fix: iCal link

* refactor: minor cleanup

* refactor: cleanup and safety check for iCal link

---------

Co-authored-by: raffi.git <51893700+rpgraffi@users.noreply.github.com>

rpgraffi committed:
(fix): course import and cleanup
